src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingUtils.java

```diff
@@ -161,7 +153,8 @@ public static boolean isStreamingSnowpipeConfigValid(final Map<String, String> i
                     "Config:{} should be present if ingestionMethod is:{}",
                     Utils.SF_ROLE,
                     inputConfig.get(INGESTION_METHOD_OPT)));
-            configIsValid = false;
+            invalidConfigParams.add(Utils.SF_ROLE);
+            // TODO @rcheng: question - can we add logging information in the error? this is confusing for customer
```

Comment:
sorry what logging information?

Reply:
The ENABLE_SCHEMATIZATION_CONFIG in Utils.java on (roughly) line 368 would be a better example. We add the enable.schematization param to the invalid params when its called on snowpipe, however we don't let the customer know that schematization is only available on streaming snowpipe. From the customer perspective, this could get confusing from just reading the error